Plugins/BuildServerIntegration/JenkinsIntegration/JenkinsAdapter.cs

```diff
@@ -267,13 +270,24 @@ private void ObserveBuilds(DateTime? sinceDate, bool? running, IObserver<BuildIn
                             return;
                         }
 
-                        var buildInfo = CreateBuildInfo((JObject)buildDetails);
-                        observer.OnNext(buildInfo);
-
-                        if (buildInfo.Status == BuildInfo.BuildStatus.InProgress)
+                        try
+                        {
+                            var buildInfo = CreateBuildInfo((JObject)buildDetails);
+
+                            if (buildInfo != null)
```

Comment:
Null check is not strictly required

Reply:
> Does it hurt?

No. But too many null checks can affect the readability. With ReSharper, you can see this better (I do not have a license.)

Build info is in this version currently set (try-catch added if failure parsing). This is a preparation for another PR that I have not decided if I will submit. I kept it as "it does not hurt".

Reply:
That's why it is preferable to invert `if`s and reduce nesting.
We'll get you a licence in few months time when the renewal time comes 
/cc: @KindDragon